### PR TITLE
Add service describe

### LINKED ISSUE #14: service describe
Write the command to describe a service! It should by default print yaml, but have a configurable printer.

### PR DESCRIPTION
**Pull request by taragu**
Fixes #14

Thanks @nimakaviani for working with me on this 🙂

### CODE REVIEW

File: pkg/kn/commands/service_describe.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"errors"
+	"github.com/spf13/cobra"
```
Comment (vdemeester) [inline pkg/kn/commands/service_describe.go]: nit: imports should be split

```go
import(
	"errors"

	"github.com/spf13/cobra"
	# […]
)

```

File: pkg/kn/commands/service_describe.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+var serviceDescribePrintFlags *genericclioptions.PrintFlags
+
+// NewServiceDescribeCommand represents the describe command
+func NewServiceDescribeCommand(p *KnParams) *cobra.Command {
+
+	serviceDescribePrintFlags := genericclioptions.NewPrintFlags("").WithDefaultOutput("yaml")
```
Comment (vdemeester) [inline pkg/kn/commands/service_describe.go]: Do we want the `describe` command to output into `yaml` by default ? (ps: it's also fine to merge like that and revise the default output format later on :angel:)

Comment (taragu) [reply pkg/kn/commands/service_describe.go]: @vdemeester I think there's some discussion going on regarding the output format https://github.com/knative/client/issues/14#issuecomment-459614902 so I plan to update this after it's finalized

Comment (cppforlife) [reply pkg/kn/commands/service_describe.go]: i've added this to our WG agenda last week.

**Comment by cppforlife** [comment]
@taragu check out comments i left in https://github.com/knative/client/pull/16. this PR should also make similar changes.

**Comment by nimakaviani** [comment]
@cppforlife, @taragu  and I had an offline chat on changes to be made to the pr. she will push an update for it.

File: pkg/kn/commands/service_describe.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import (
+	"errors"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+// NewServiceDescribeCommand represents the describe command
```
Comment (nimakaviani) [inline pkg/kn/commands/service_describe.go]: not sure if the comment delivers any value here.

File: pkg/kn/commands/service_describe_test.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	"k8s.io/apimachinery/pkg/api/equality"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	serving "github.com/knative/serving/pkg/client/clientset/versioned/typed/serving/v1alpha1"
+	"github.com/knative/serving/pkg/client/clientset/versioned/typed/serving/v1alpha1/fake"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	client_testing "k8s.io/client-go/testing"
+)
+
+func fakeDescribe(args []string, response *v1alpha1.Service) (action client_testing.Action, output string, err error) {
```
Comment (nimakaviani) [inline pkg/kn/commands/service_describe_test.go]: qualify the name, maybe `fakeServiceDescribe`?

File: pkg/kn/commands/service_describe_test.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	cmd.SetArgs(args)
+	err = cmd.Execute()
+	if err != nil {
+		return
+	}
+	output = buf.String()
+	return
+}
+
+func TestDescribeEmpty(t *testing.T) {
```
Comment (nimakaviani) [inline pkg/kn/commands/service_describe_test.go]: qualify the name `TestEmptyServiceDescribe`

File: pkg/kn/commands/service_describe_test.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+	_, _, err := fakeDescribe([]string{"service", "describe"}, &v1alpha1.Service{})
+	if err == nil {
+		t.Fatal("Should expect error")
+	}
+	if !strings.Contains(err.Error(), "Please provide one service name to describe.") {
+		t.Fatalf("Expect error 'Please provide one service name to describe', but got %v", err)
+	}
+}
+
+func TestDescribeDefaultOutput(t *testing.T) {
```
Comment (nimakaviani) [inline pkg/kn/commands/service_describe_test.go]: qualify the name `TestServiceDescribeDefaultOutput`

File: pkg/kn/commands/service_describe.go [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+		Use:   "describe NAME",
+		Short: "Describe available services.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := p.ServingFactory()
+			if err != nil {
+				return err
+			}
+
+			if len(args) < 1 {
+				return errors.New("Please provide one service name to describe. ")
```
Comment (nimakaviani) [inline pkg/kn/commands/service_describe.go]: have some consistency for error messages between different describe commands (revision and service in this case)

**Pull request merged**